src/bsoncxx/test/bson_value.cpp

```diff
@@ -54,6 +54,28 @@ TEST_CASE("types::bson_value::value", "[bsoncxx::types::bson_value::value]") {
         SECTION("bool") {
             auto test_doc = bson_value::make_value(types::b_bool{true});
             value_construction_test(test_doc.view());
+
+            SECTION("with explicit BSON type") {
+                bson_value::value copy_initialization = types::b_bool{true};
+                REQUIRE(copy_initialization == test_doc);
+
+                bson_value::value direct_initialization(types::b_bool{true});
+                REQUIRE(direct_initialization == test_doc);
+
+                bson_value::value _static_cast = (bson_value::value)types::b_bool{true};
```

Comment:
Technically, this is not a static cast, this is a c-style cast.  I would suggest either changing the name or changing the cast, as it's misleading the way it's written:

`bson_value::value cast = (bson_value::value)types::b_bool{true};`
`bson_value::vale _static_cast = static_cast<bson_value::value>(types::b_bool{true});`

Reply:
Done